code/game/objects/items/breakable.dm

```diff
@@ -0,0 +1,129 @@
+//Breakable items
+
+//TODO:
+//multiples of a single fragment
+//add damaged text into the general examine proc
+//check slimes
+//add thrown response
+//projectiles?
+//kicking and biting?
+//change "hit" to something else?
+//hit sounds
+//punching hurting hands and stuff?
+//use kicking code
+//generalize out of devices
+
+/obj/item/device
+
+	//Destructability parameters:
+	var/breakable = 0 //1: breakable (by smashing it using the following flags
+	var/breakable_flags = 0 /*possible flags include BREAKABLE_ALL | BREAKABLE_HIT | BREAKABLE_HIT_EMPTY | BREAKABLE_HIT_WEAPON | BREAKABLE_THROW
+							BREAKABLE_HIT encompasses both BREAKABLE_HIT_EMPTY and BREAKABLE_HIT_WEAPON */
+	var/struct_integ = 15 //structural integrity of the item, akin to HP.
+	var/struct_integ_max = 15
+	var/damage_armor = 5 //attacks of this much damage or below will glance off
+	var/damage_resist = 5 //attacks stronger than damage_armor will have their damage reduced by this much
+	var/damaged_text = "" //Addendum to the description when it's damaged eg. damaged_text of "It is dented." Empty string "" will skip this addendum.
+	var/breaks_text = "" //Visible message when the items breaks. eg. "breaks apart" Empty string skips this.
+	var/breaks_sound = "" //path to audible sound when the item breaks. Empty string skips this.
+	var/list/breakable_fragments = list() //List of objects that will be produced when the item is broken apart. eg. /obj/weapon/item/shard
+
+/obj/item/device/proc/on_broken() //Called right before an object breaks
+	if(breaks_text!="")
+		visible_message("<span class='notice'>\The [src] [breaks_text]!</span>")
+	if(breaks_sound!="")
+		playsound(src, breaks_sound, 50, 1)
+	if(breakable_fragments.len)
+		drop_fragments()
+
+/obj/item/device/proc/drop_fragments() //Separate proc in case special stuff happens with a given item's fragments
+	if(breakable_fragments.len)
+		for(var/i in 1 to breakable_fragments.len)
+			var/obj/item/thisfragment=breakable_fragments[i]
+			new thisfragment(get_turf(src))
+
+/obj/item/device/proc/receive_damage(var/incoming_damage)
+	var/thisdmg=(incoming_damage>max(damage_armor,damage_resist)) * (incoming_damage-damage_resist) //damage is 0 if the incoming damage is less than either damage_armor or damage_resist, to prevent negative damage by weak attacks
+	struct_integ-=thisdmg
+	if(struct_integ<=0)
+		on_broken()
+		qdel(src)
+	if(!thisdmg)
+		return 0 //return 0 if the item took no damage (glancing attack)
+	else
+		return 1 //return 1 if the item took damage
+
+/////////////////////
+
+
+//Breaking items:
+
+//Aliens
+/obj/item/device/attack_alien(mob/living/carbon/alien/humanoid/user)
+	if(user.a_intent == I_HURT && breakable && breakable_flags | BREAKABLE_HIT_EMPTY)
```

Comment:
The correct syntax is `value & FLAG_TO_CHECK`. As it is right now, `breakable_flags | BREAKABLE_HIT_EMPTY` would always be true.